### PR TITLE
perf: upgrade zustand to v5 

### PR DESCRIPTION
**Pull request by RonenMars**
## Description
Optimize our app with the latest release:
https://github.com/pmndrs/zustand/releases/tag/v5.0.0

Mostly: Drop many old things, which will improve our app performance.

Comes from dependabot: https://github.com/autokitteh/web-platform/pull/545

## Linear Ticket
https://linear.app/autokitteh/issue/UI-827/upgrade-app-to-zustand-v5

## What type of PR is this? (check all applicable)

-   [ ] 💡 (feat) - A new feature (non-breaking change which adds functionality)
-   [ ] 🔄 (refactor) - Code Refactoring - A code change that neither fixes a bug nor adds a feature
-   [ ] 🐞 (fix) - Bug Fix (non-breaking change which fixes an issue)
-   [x] 🏎 (perf) - Optimization
-   [ ] 📄 (docs) - Documentation - Documentation only changes
-   [ ] 📄 (test) - Tests - Adding missing tests or correcting existing tests
-   [ ] 🎨 (style) - Styles - Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
-   [ ] ⚙️ (ci) - Continuous Integrations - Changes to our CI configuration files and scripts (example scopes: Travis, Circle, BrowserStack, SauceLabs)
-   [ ] ☑️ (chore) - Chores - Other changes that don't modify src or test files
-   [ ] ↩️ (revert) - Reverts - Reverts a previous commit(s).

### CODE REVIEW

File: src/components/molecules/drawer.tsx [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
 import { useDrawerStore } from "@src/store";
 import { cn } from "@src/utilities";
 
 export const Drawer = ({ children, className, name, variant }: DrawerProps) => {
-	const { isOpen, onClose } = useDrawerStore((state) => ({
-		isOpen: state.drawers[name],
-		onClose: state.closeDrawer,
-	}));
+	const { isOpen, onClose } = useDrawerStore(
+		useShallow((state) => ({
```
[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/molecules/drawer.tsx:12]: In the official documentation: https://zustand.docs.pmnd.rs/migrations/migrating-to-v5 they recommend to use it in the components.

[the remark this answers is not in the document]
Comment (J1za) [reply src/components/molecules/drawer.tsx:12]: I don't see a recommendation, only how migration from v4 to v5 and use `useShallow`. And we need must be called  `useShallow` every time in all components, it's not convenient.

**Review by J1za — APPROVED** [review]

File: src/components/molecules/drawer.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (J1za) [reply src/components/molecules/drawer.tsx]: Cool

[the remark this answers is not in the document]
Comment (RonenMars) [reply src/components/molecules/drawer.tsx]: Changed all in:
https://github.com/autokitteh/web-platform/pull/586/commits/2dfcdafbac23fe8f643131a22491903127765995

**Pull request merged**